ReactWindows/ReactNative.Core/Bridge/GenericDelegateTypes.cs

```diff
@@ -0,0 +1,534 @@
+        }
+
+        public object Invoke(object[] args)
+        {
+            _instancedDelegate((T0)args[0], (T1)args[1], (T2)args[2], (T3)args[3], (T4)args[4], (T5)args[5]);
+            return null;
+        }
+
+        public object DelegateTarget
+        {
+            get
+            {
+                return _instancedDelegate.Target;
+            }
+        }
+    }
+
+    public class SignatureAction7<T0, T1, T2, T3, T4, T5, T6> : IGenericDelegate
+    {
+        Action<T0, T1, T2, T3, T4, T5, T6> _instancedDelegate;
+
+        public SignatureAction7(object instance, MethodInfo method)
+        {
+            _instancedDelegate = (Action<T0, T1, T2, T3, T4, T5, T6>)method.CreateDelegate(typeof(Action<T0, T1, T2, T3, T4, T5, T6>), instance);
+        }
+
+        public object Invoke(object[] args)
+        {
+            _instancedDelegate((T0)args[0], (T1)args[1], (T2)args[2], (T3)args[3], (T4)args[4], (T5)args[5], (T6)args[6]);
+            return null;
+        }
+
+        public object DelegateTarget
+        {
+            get
+            {
+                return _instancedDelegate.Target;
+            }
+        }
+    }
+
+    public class SignatureAction8<T0, T1, T2, T3, T4, T5, T6, T7> : IGenericDelegate
+    {
+        Action<T0, T1, T2, T3, T4, T5, T6, T7> _instancedDelegate;
+
+        public SignatureAction8(object instance, MethodInfo method)
+        {
+            _instancedDelegate = (Action<T0, T1, T2, T3, T4, T5, T6, T7>)method.CreateDelegate(typeof(Action<T0, T1, T2, T3, T4, T5, T6, T7>), instance);
+        }
+
+        public object Invoke(object[] args)
+        {
+            _instancedDelegate((T0)args[0], (T1)args[1], (T2)args[2], (T3)args[3], (T4)args[4], (T5)args[5], (T6)args[6], (T7)args[7]);
+            return null;
+        }
+
+        public object DelegateTarget
+        {
+            get
+            {
+                return _instancedDelegate.Target;
+            }
+        }
+    }
+
+    public class SignatureAction9<T0, T1, T2, T3, T4, T5, T6, T7, T8> : IGenericDelegate
+    {
+        Action<T0, T1, T2, T3, T4, T5, T6, T7, T8> _instancedDelegate;
+
+        public SignatureAction9(object instance, MethodInfo method)
+        {
+            _instancedDelegate = (Action<T0, T1, T2, T3, T4, T5, T6, T7, T8>)method.CreateDelegate(typeof(Action<T0, T1, T2, T3, T4, T5, T6, T7, T8>), instance);
+        }
+
+        public object Invoke(object[] args)
+        {
+            _instancedDelegate((T0)args[0], (T1)args[1], (T2)args[2], (T3)args[3], (T4)args[4], (T5)args[5], (T6)args[6], (T7)args[7], (T8)args[8]);
+            return null;
+        }
+
+        public object DelegateTarget
+        {
+            get
+            {
+                return _instancedDelegate.Target;
+            }
+        }
+    }
+
+    public class SignatureAction10<T0, T1, T2, T3, T4, T5, T6, T7, T8, T9> : IGenericDelegate

```

Comment:
Since you'll be generating this code, there's no reason to stop at 10, you might as well generate all 16 that are available.